### PR TITLE
DVX-338: Adds `EnumDef` update method to add new `ElementDefs`

### PR DESCRIPTION
**Pull request by Aryamanz29**

### CODE REVIEW

File: pyatlan/cache/enum_cache.py [hunk 1]
```diff
@@ -33,14 +33,17 @@ def refresh_cache(cls) -> None:
...
 
-        :param name: human-readable name of the enumeration
-        :returns: the enumeration definition
+        :param name: human-readable name of the enumeration.
+        :raises `NotFoundError`: if the enumeration with the given name does not exist.
+        :returns: enumeration definition
         """
-        return cls.get_cache()._get_by_name(name=name)
+        if not (enum := cls.get_cache()._get_by_name(name=name)):
+            raise ErrorCode.ENUM_NOT_FOUND.exception_with_parameters(name)
```
Comment (cmgrote) [inline pyatlan/cache/enum_cache.py:45]: I'm OK with this, as I don't anticipate it'll be a heavily-used area. However, technically this is a breaking change in the interface. Let's just be sure that in the next release notes we indicate this change in behavior for this particular method as a breaking change?

**Review by ErnestoLoma — APPROVED** [review]
Looks fine

**Review by cmgrote — APPROVED** [review]
See comment related to next set of release notes — but looks good to merge.

File: pyatlan/cache/enum_cache.py (hunk 1 above)
Comment (Aryamanz29) [reply pyatlan/cache/enum_cache.py:45]: Noted!

**Pull request merged**